### PR TITLE
Physics3d using bullet 2.82

### PR DESCRIPTION
**Pull request by super626**

### CODE REVIEW

File: tests/cpp-tests/Resources/ccs-res [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit afa5ba56b3745b0fb94e730ea54a79e598c31ba8
+Subproject commit 7b3f21b5ad03445b907e0b713804fdc35f3c0959
```
Comment (minggo) [inline tests/cpp-tests/Resources/ccs-res]: Why update this submodule?

**Comment by minggo** [comment]
Could you please
- undo submodule modification
- fix conflict
- update external/config.json to bullet, it is merged in 3rd party repo

**Comment by minggo** [comment]
Oh, don't have to update external/config.json, it is already updated.

**Comment by super626** [comment]
Thanks @minggo. done

File: cocos/3d/CCTerrain.h [hunk 2]
```diff
@@ -384,6 +384,17 @@ class CC_DLL Terrain :public Node
...
+    
+    /**
+     * get the terrain's size
+     */
+    Size getTerrainSize() const { return Size(_imageWidth, _imageHeight); }
+    
+    /**
+     * get the terrain's height data
+     */
+    void getHeightData(std::vector<float>& data) const;
```
Comment (minggo) [inline cocos/3d/CCTerrain.h]: Why return value in parameter?
It doesn't obey coding style, and it is hard for script binding codes to implement it.

Comment (super626) [inline cocos/3d/CCTerrain.h]: I have modified to  void getTerrainSize(Size& size)

Comment (minggo) [inline cocos/3d/CCTerrain.h]: May be you misunderstood me.
I mean not return value in parameter.

**Comment by super626** [comment]
[ci rebuild]

**Comment by super626** [comment]
[ci rebuild]

**Comment by super626** [comment]
[ci rebuild]

**Comment by minggo** [comment]
[ci rebuild]

**Comment by minggo** [comment]
And please fix conflict because VS2012 support is removed.

**Comment by super626** [comment]
[ci rebuild]

**Comment by super626** [comment]
[ci rebuild]

**Pull request closed without merging**